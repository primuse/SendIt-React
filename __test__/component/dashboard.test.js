import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Dashboard from '../../src/components/dashboard.jsx';

global.Headers = () => {};

const initialState1 = {
  auth: {
    isAuthenticated: true,
    user: {
      isadmin: false,
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com'
    },
  },
  users: {
    allUsers: [],
    show: false
  },
  parcel: {
    parcels: [
      {
        id: 1,
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: 2,
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    pages: 2
  }
};

const initialState2 = {
  auth: {
    isAuthenticated: true,
    user: {
      isadmin: true,
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com'
    },
  },
  users: {
    allUsers: [],
    show: false
  },
  parcel: {
    parcels: [
      {
        id: 1,
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: 2,
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    pages: 2
  }
};

const mockStore = configureStore([thunk]);


describe('All user parcels component', () => {
  const store = mockStore(initialState1);
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('All parcels component', () => {
  const store = mockStore(initialState2);
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
