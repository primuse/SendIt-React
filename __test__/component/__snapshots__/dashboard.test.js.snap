// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All users component renders correctly 1`] = `
<main>
  <aside
    className=""
  >
    <div
      className="bg-blue"
      id="menu"
    >
      <div
        id="aside-profile"
      >
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          <h3
            className="m-0 pb-10"
          >
            Send
            <span
              className="head-span"
            >
              IT
            </span>
          </h3>
        </a>
        <small
          className="d-block pb-10 white-3"
          id="user-id"
        >
          SD0
        </small>
        <small>
          <p
            className="m-0"
          >
            Welcome 
          </p>
          <p
            id="username"
          >
             
          </p>
        </small>
      </div>
      <ul
        className="mt-80"
      >
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            <span>
              Dashboard
            </span>
          </a>
        </li>
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/delivered_parcels"
            onClick={[Function]}
          >
            <span>
              Delivered Parcels
            </span>
          </a>
        </li>
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/inTransit_parcels"
            onClick={[Function]}
          >
            <span>
              In-transit Parcels
            </span>
          </a>
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="grey"
    id="dash"
  >
    <nav>
      <a
        className="toggler mr-1"
        href="#"
        id="toggle"
        onClick={[Function]}
      >
        <img
          src="test-file-stub"
        />
      </a>
      <ul
        className="p-0"
      >
        <li>
          Dashboard
        </li>
      </ul>
      <ul
        className="p-0 ml-auto"
      >
        <li
          className="mr-1"
        >
          <a
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
    <div
      id="main-content-page"
    >
      <div
        className="mb-34"
        id="info-cont"
      >
        <div
          className="info-box"
        >
          <small>
            Created
          </small>
          <h4
            className="m-0"
            id="created"
          >
            0
          </h4>
        </div>
        <div
          className="info-box"
        >
          <small>
            In Transit
          </small>
          <h4
            className="m-0 blue"
            id="in-transit"
          >
            0
          </h4>
        </div>
        <div
          className="info-box"
        >
          <small>
            Delivered
          </small>
          <h4
            className="m-0 green"
            id="delivered"
          >
            0
          </h4>
        </div>
      </div>
      <div
        className="dash-cont"
      >
        <div
          id="table-info"
        >
          <a
            href="#"
          >
            All
          </a>
        </div>
        <div
          id="table-cont"
        >
          <table>
            <tbody>
              <tr
                id="table-head"
              >
                <th>
                  ID
                </th>
                <th>
                  Name
                </th>
                <th>
                  Weight
                </th>
                <th>
                  Price
                </th>
                <th>
                  Destination
                </th>
                <th>
                  Receiver
                </th>
                <th>
                  Sent On
                </th>
                <th>
                  Status
                </th>
              </tr>
              <tr>
                <td>
                  PO
                  1
                </td>
                <td />
                <td />
                <td>
                  4,000
                </td>
                <td>
                  Owerri
                </td>
                <td>
                  Tiku Okoye
                </td>
                <td>
                  22-03-2019
                </td>
                <td>
                  created
                </td>
                <td>
                  <a
                    className="btn xsm bg-bright-blue white"
                    href="/details/1"
                    onClick={[Function]}
                  >
                    View
                  </a>
                </td>
              </tr>
              <tr>
                <td>
                  PO
                  2
                </td>
                <td />
                <td />
                <td>
                  4,000
                </td>
                <td>
                  Owerri
                </td>
                <td>
                  Tiku Okoye
                </td>
                <td>
                  22-03-2019
                </td>
                <td>
                  created
                </td>
                <td>
                  <a
                    className="btn xsm bg-bright-blue white"
                    href="/details/2"
                    onClick={[Function]}
                  >
                    View
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </section>
</main>
`;
